Source/core/MessageStore.cpp

```diff
@@ -102,9 +104,23 @@ ENUM_CONVERSION_END(Core::Messaging::Metadata::type)
                 _adminLock.Unlock();
             }
 
+            void SetStorage(Core::Messaging::IStore* storage)
+            {
+                ASSERT((_storage == nullptr) ^ (storage == nullptr));
+
+                _adminLock.Lock();
+                _storage = storage;
+                _adminLock.Unlock();
+            }
+
+            Core::Messaging::IStore* Storage() const {
+                return (_storage);
+            }
```

Comment:
_storage is not properly protected here. 
But it doesn't seem to me that we need to export Storage out of the controls instance. It is only required during Announce which is also on controls? Then the Announce should use it's insternal _storage and call Default and it already can guard the nullptr.